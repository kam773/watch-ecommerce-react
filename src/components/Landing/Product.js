import React from "react";
import styled from "styled-components";
import { ProductConsumer } from "../../context";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import { Card } from "reactstrap";

export default class Product extends React.Component {
  render() {
    const { id, title, img, price } = this.props.product;
    return (
      <ProductWrapper className="col-9 mx-auto col-md-6 col-lg-3 my-3">
        <Card>
          <ProductConsumer>
            {value => (
              <div
                className="img-container p-5"
                onClick={() => value.handleDetail(id)}
              >
                <Link to="/details">
                  <img src={img} alt="" className="card-img-top" />
                </Link>
              </div>
            )}
          </ProductConsumer>
          {/* Card footer */}
          <div className="card-footer d-flex justify-content-between">
            <p className="align-self-center mb-0"> {title} </p>
            <h5 className="text-blue font-italic mb-0">
              <span className="mr-1"> $ {price} </span>
            </h5>
          </div>
        </Card>
      </ProductWrapper>
    );
  }
}

Product.propTypes = {
  product: PropTypes.shape({
    id: PropTypes.number,
    img: PropTypes.string,
    title: PropTypes.string,
    price: PropTypes.number,
    inCart: PropTypes.bool
  }).isRequired
};

const ProductWrapper = styled.div`
  .cart-btn {
    background: transparent;
    border: none;
    cursor: pointer;
    padding: 0.8rem;
    font-size: 1.5rem;
    position: absolute;
    bottom: 0;
    right: 0;
  }
  .img-container {
    position: relative;
    overflow: hidden;
  }
  .card-img-top {
    transition: all 0.5s linear;
  }
  .img-container:hover .card-img-top {
    transform: scale(1.2);
  }
`;
