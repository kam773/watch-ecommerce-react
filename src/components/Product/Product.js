import React from 'react'
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { ProductConsumer } from '../../context';
import PropTypes from 'prop-types';

export default class Product extends React.Component {
  render() {
   const { id, title, img, price, inCart } = this.props.product
    return (
      <ProductWrapper className="col-9 mx-auto col-md-6 col-lg-3 my-3">
          <div className="card">
            <ProductConsumer>
            { (value) => ( <div className="img-container p-5"
                           onClick={ () =>
                              value.handleDetail(id) }>
                        <Link to="/details">
                            <img src={ img } alt="" className="card-img-top"/>
                        </Link>
                        <button
                        className="cart-btn"
                        disabled={ inCart? true : false }
                        onClick={ ()=>{
                              value.addToCart(id);
                              value.openModal(id);

                        } }>
                            { inCart ?
                      (<p className="text-capitalize mb-0" disabled>
                          {" "}
                      in cart
                      </p>) : (
                          <i className="fas fa-cart-plus"></i>
                      )}
                        </button>
                    </div>) }
              </ProductConsumer>
              {/* Card footer */}
              <div className = "card-footer d-flex justify-content-between">
                            <p className = "align-self-center mb-0"> {
                        title
                      } </p>
                            <h5 className = "text-blue font-italic mb-0" >
                                <span className = "mr-1" > $ {
                        price
                      } </span></h5>
              </div>

          </div>

      </ProductWrapper>
    )
  }
}

Product.propTypes = {
  product: PropTypes.shape({
      id: PropTypes.number,
      img: PropTypes.string,
      title: PropTypes.string,
      price: PropTypes.number,
      inCart: PropTypes.bool
  }).isRequired
}

const ProductWrapper = styled.div`
  .cart-btn {
    background: transparent;
    border: none;
    cursor: pointer;
    padding: .8rem;
    font-size: 1.5rem;
    position: absolute;
    bottom: 0;
    right: 0;
  }
  .img-container {
    position: relative;
    overflow: hidden;
  }
  .card-img-top {
    transition: all .5s linear;
  }
  .img-container:hover .card-img-top {
    transform: scale(1.2);
  }

`;