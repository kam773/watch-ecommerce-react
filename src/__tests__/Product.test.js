import React from 'react';
import Product from '../components/Product/Product';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router'
import { Link } from 'react-router-dom';

describe("Product", () => {
    it("renders without crashes", () => {
        const props = {
            product: {}
        }
        const tree = renderer.create(
            <StaticRouter>
                <Product {...props}/>
            </StaticRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('has div with class', () => {
        const props = {
            product: {}
        };
        const wrapper = shallow(<Product {...props}/>)
        expect(wrapper.find('.card').exists()).toEqual(true);
    });
    it('should contain  link', () => {
        const props = {
            product: {}
        };
        const wrapper = shallow(<Product {...props}/>);
        expect(wrapper.find(Link).filter({ to: '/details' })).toBeTruthy();
    });
    it('has div with class', () => {
        const props = {
            product: {}
        };
        const wrapper = shallow(<Product {...props}/>)
        expect(wrapper.find('.card-footer').exists()).toEqual(true);
    });
    it('has p element', () => {
        const props = {
            product: {}
        };
        const wrapper = shallow(<Product {...props}/>)
       expect(wrapper.find('p').exists()).toEqual(true);
    });
    it('has img element', () => {
        const props = {
            product: {}
        };
        const wrapper = shallow(<Product {...props}/>);
        const post = wrapper.find('h5');
        expect(['text-blue', 'font-italic', 'mb-0'].every(c => post.hasClass(c))).toEqual(true);
    });
})