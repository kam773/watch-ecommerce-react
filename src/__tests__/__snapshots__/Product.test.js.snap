// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product renders without crashes 1`] = `
<div
  className="col-9 mx-auto col-md-6 col-lg-3 my-3 sc-bdVaJa xNYtp"
>
  <div
    className="card"
  >
    <div
      className="img-container p-5"
      onClick={[Function]}
    >
      <a
        href="/details"
        onClick={[Function]}
      >
        <img
          alt=""
          className="card-img-top"
        />
      </a>
      <button
        className="cart-btn"
        disabled={false}
        onClick={[Function]}
      >
        <i
          className="fas fa-cart-plus"
        />
      </button>
    </div>
    <div
      className="card-footer d-flex justify-content-between"
    >
      <p
        className="align-self-center mb-0"
      >
         
         
      </p>
      <h5
        className="text-blue font-italic mb-0"
      >
        <span
          className="mr-1"
        >
           $ 
           
        </span>
      </h5>
    </div>
  </div>
</div>
`;
